resource/graph_node.go

```diff
@@ -26,12 +25,14 @@ const (
 	// NodeStateConfiguring denotes a resource is being configured.
 	NodeStateConfiguring
```

Comment:
Did we ever resolve @dgottlieb's concern? Can a resource be both usable and not usable in `NodeStateConfiguring`?

Reply:
Not really, I think making the "configuring" start more precise is a future improvement. Right now this state mostly mirrors the old `needsConfiguration` boolean field.